### PR TITLE
Adding support for internet facing elb.

### PR DESCRIPTION
**Pull request by ajaijoy-bmc**
1. Updated model in UpsertAmazonLoadBlancerDescription with isInternal flag.
2. Updated the aws sdk interaction to pass in scheme= internal only if isInternal flag is set

### CODE REVIEW

**Comment by cfieber** [comment]
thanks for this! there is a failing test in UpsertAmazonLoadBalancerAtomicOperationSpec ( https://travis-ci.org/spinnaker/clouddriver/builds/106849806 ) but otherwise this is a nice change

**Comment by ajaijoy-bmc** [comment]
Thanks. I fixed the unit test, it should show up now.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy [hunk 1]
```diff
@@ -76,6 +76,8 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
       def regionScopedProvider = regionScopedProviderFactory.forRegion(description.credentials, region)
       def loadBalancerName = description.name ?: "${description.clusterName}-frontend".toString()
 
+      boolean isInternal = description.isInternal == null ? false : description.isInternal.toBoolean();
```
Comment (cfieber) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy]: could you update this: such that if isInternal == null it falls back to checking if description.subnetType.contains('internal')

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperationSpec.groovy [hunk 2]
```diff
@@ -244,6 +244,7 @@ class UpsertAmazonLoadBalancerAtomicOperationSpec extends Specification {
 
   void "should handle VPC ELB creation"() {
       description.subnetType = "internal"
+      description.isInternal = true;
```
Comment (cfieber) [inline clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperationSpec.groovy]: could you add a test where isInternal is null and subnetType is 'internal' to confirm we have maintained backwards compatibility

**Comment by claymccoy** [comment]
It looks like this just needs the changes that @cfieber mentioned for backwards compatibility.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy (hunk 1 above)
Comment (ajaijoy) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy]: yes, will work on this today

File: clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperationSpec.groovy (hunk 2 above)
Comment (ajaijoy) [inline clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperationSpec.groovy]: sure will do today

**Comment by ajordens** [comment]
Looks like we're still waiting on a couple updates.

**Comment by ajaijoy** [comment]
@ajordens  This pull request is already taken care of. I had raised another pull request because of an accidental branch delete.

**Comment by ajordens** [comment]
Ahh yes. 

Sorry, I was just blindly going through the old PRs and comment.

I'll close it.
